lib/src/xml/grammar.dart

```diff
@@ -38,22 +38,25 @@ abstract class XmlGrammarDefinition<TNode, TName>
   // Connects the productions and the XML AST callbacks.
 
   @override
-  Parser attribute() => super
-      .attribute()
-      .map((each) => createAttribute(each[0], each[4][0], each[4][1]));
+  Parser attribute() => super.attribute().map((each) {
+        each as List<dynamic>;
+        return createAttribute(
+            each[0], (each[4] as List)[0], (each[4] as List)[1]);
+      });
 
   @override
   Parser attributeValueDouble() => super
       .attributeValueDouble()
-      .map((each) => [each[1], XmlAttributeType.DOUBLE_QUOTE]);
+      .map((each) => [(each as List)[1], XmlAttributeType.DOUBLE_QUOTE]);
```

Comment:
changes like this are so common, perhaps a method should be added to Parser? or a static extension?

Reply:
Yeah, this is very common for all PetitParser grammars. How would such an extension look like? 

I hope once code generation becomes more streamlined we can created properly typed parsers all the way down.

Reply:
I hadn't come up with a great extension off the cuff. 😛  but something like:

```
extension MapParserExtension<T> on Parser<T> {
  Parser<R> mapAsList<R>(Callback<List<Object?>, R> callback, {bool hasSideEffects = false}) =>
      MapParser<T, R>(this, (each) => callback(each as List<Object?>, hasSideEffects);
}
```

Reply:
castList for the win!